geps/gep-713/index.md

```diff
@@ -1,1157 +1,743 @@
 
-However, it relies on the establishment and communication of a convention ("An 
-EffectivePolicy is right next to your affected object"), that may not be desirable.
+In the context of traffic networking, for example, often the question asked by users is *“What happens when a network request X comes in?”*, stated in terms of a concrete “X”. There is an implicit expectation that a set of Kubernetes resources suffices to represent all the rules for a given workload to be activated and thus process request X, and often that is the case. For more complex cases however (multiple personas, application concerns separated into dedicated resource kinds, interaction between groups of users, etc), real life can get more complicated than a simple `kubectl get x`, or at least additional steps must be automated to encompass complexity into what can be achieved with a single declarative object.
 
-Thus its status as EXPERIMENTAL DO NOT USE YET.
+With that in mind, a possible solution for the discoverability problem may involve designing tools (e.g. CLI tools/plugins), new CRDs, etc that let users ask questions in terms of the real life problems they have to deal with on a daily basis, rather than shaped by the underlying technologies used in the process. For instance, a simple Kubernetes object that is used to declare the rules to process a HTTP request cannot have its status reported simply as Ready/Not ready. By being a complex object composed of multiple routing rules, potentially affected by specifications declared from other objects as well, its status MUST account for that complexity and be structured in such a way that informs the owner with respect to each possible case, whether the ones induced by the internal specification declared by the object itself or its external relationships.
 
-#### Validating Admission Controller to inform users about relevant Policy
+In other words, the discoverability problem exists and must be addressed in light of the complexity associated with the topology of contexts induced by a set of hierarchically related resources. One should always have that topology in mind while asking questions regarding the behavior of a given resource, because just like a routing object (e.g. HTTPRoute) does not exist independently from its parent contexts (e.g. Gateways) or its children (e.g. Backends), any resource in focus may be just a part of a whole.
 
-Implementations MAY supply a Validating Admission Webhook that will return a
-WARNING message when an applied object is affected by some Policy.
+### Status reporting
 
-The warning message MAY include the name, namespace, apiGroup and Kind of relevant
-Policy objects.
+#### Meta resource status
 
-##### Design Considerations
+Meta resource CRDs should define a status stanza that allows for reporting the status of the meta resource with respect to each context the resource may apply.
 
-Pro:
+The basic status conditions are:
```

Comment:
what does this mean? an implementation should or must have these? Or if they want similar concepts they should consider these names for consistency

Reply:
This basically means I'm a coward 😜 I didn't want to push for a particular closed set of resource/policy status conditions, but to take a step back and collect more feedback first.

What you see here is my take on what can be done to enhance the current proposed conditions (which only has one prescribed condition really – i.e.,`Accepted`), and that is still generic enough IMO that all implementations could consider implementing (a bold guess of mine perhaps).

Do you have an opinion on MUST, SHOULD, MAY (...want to consider adopting similar concepts)? Or even on the list of conditions itself?

Reply:
In general I think for policy, given its an inherently vendor-specific concept, should probably have few, if any, MUSTs. Its not enforceable anyways (unless we have some very creative conformance tests, I guess).

I think having a set of standardized conditions that you MAY apply is likely good. That way any implementation declaring a concept like "the config is valid" will use `Accepted` vs some different terms,

Reply:
I think it's worth having some things as SHOULD to indicate that they're a really good idea and if you're not going to, you need to know why.

Reply:
Until we have a better solution, changed to:
- Metaresource status
  - "CRDs MUST define a status stanza"
  - "implementations SHOULD support the basic status conditions"
- Target object status
  - "Implementations SHOULD put a condition"
  - "condition, if added, MUST be named according to the pattern"
  - "Implementations SHOULD use their own unique domain prefix"
  - "For objects that do not have a `status.Conditions` field […], object SHOULD instead have an annotation"